src/features/auth/SignUp.jsx

```diff
@@ -0,0 +1,190 @@
+import { userSignUp } from "./authSlice";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+export const SignUp = () => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [userName, setUserName] = useState("");
+  const [errors, setErrors] = useState({});
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const token = useSelector((state) => state.auth.token);
+
+  const signUpStatus = useSelector((state) => state.auth.signUpStatus);
+  useEffect(() => {
+    if (signUpStatus === "succeeded") {
+      localStorage?.setItem("login", JSON.stringify({ token }));
+      navigate("/feed");
+    }
+  }, [signUpStatus, navigate, token]);
+
+  // useEffect(() => {
+  //   if (loggedInUserStatus === "succeeded") {
+  //     navigate("/feed");
+  //   }
+  // }, [loggedInUserStatus]);
+
+  const signUp = async () => {
+    if (signUpStatus === "idle" || signUpStatus === "failed") {
+      dispatch(userSignUp({ email, firstName, lastName, userName, password }));
+    }
+  };
+
+  function handleSignUp(e) {
+    e.preventDefault();
+    // console.log("helo");
+    const validationError = validateSignUp(
+      firstName,
+      lastName,
+      userName,
+      email,
+      password
+    );
+    if (Object.keys(validationError).length > 0) {
+      return setErrors(validationError);
+    }
+    signUp({ firstName, lastName, userName, email, password });
+  }
+
+  function validateSignUp(firstName, lastName, userName, email, password) {
```

Comment:
You can move these to a helper file so that you can use some of them in your login also, like email or password validation, anyway, your code will be more modular then.